### PR TITLE
Batcher: Flushing

### PR DESCRIPTION
**Pull request by brianshannan-wf**
Added a flush function to forcibly complete a batch and made Dispose() automatically flush the current batch before disposing. I also changed Get() to return a disposed error iff the batcher is disposed and there are no more ready batches. Added a few extra checks to prevent possible problems.

@tylertreat-wf @stevenosborne-wf @matthewgardner-wf @dustinhiatt-wf @alexandercampbell-wf

### CODE REVIEW

File: batcher/batcher.go [hunk 1]
```diff
@@ -117,16 +134,50 @@ func (b *basicBatcher) Get() ([]interface{}, error) {
...
 func (b *basicBatcher) Dispose() {
 	b.lock.Lock()
+	if b.disposed {
+		b.lock.Unlock()
+		return
+	}
+	b.flush()
 	b.disposed = true
 	b.items = nil
 	close(b.batchChan)
```
Comment (dustinhiatt-wf) [inline batcher/batcher.go]: Does calling close on a closed channel panic?

**Comment by dustinhiatt-wf** [comment]
+1

**Comment by tylertreat-wf** [comment]
+1

**Comment by alexandercampbell-wf** [comment]
+1

**Pull request merged**